src/sparse/vec.rs

```diff
@@ -1086,6 +1086,7 @@ impl<'a, 'b, N, I, Iptr, IpS1, IS1, DS1, IS2, DS2>
     for &'a CsMatBase<N, I, IpS1, IS1, DS1, Iptr>
 where
     N: Clone
+        + Copy
```

Comment:
Why do you add `Copy` here?

Reply:
right someimtes I can't even quite follow exactly when it needs Copy or not. I was just fixing compile errors. I figure there might be scope for a T:MatrixElem kind of shared bound to put in these places (do we need Clone / Zero / Copy or what by default).   It did seem like with MulAcc::mul(&a,&b)  "copy" shouldn't be needed. .. but sometimes maths libs make an assumption that the component types are trivially copyable (I am writing heavily generic 'graphics' type vecmaths myself elsewhere, i've bounced back and forth on this myself)

let me go over some of the other reqested changes.. were you able to integrate them yourself (then I can grab them) or shall i replicate them ? I'm happy with all those updates myself.

> This break type inference, as we see in our tests. Could we introduce dot_acc as an accumulating dot product and leave dot as MulAcc<N, N>?

I'll have to think more about this possiblitiy .. do you mean a seperate   `my_accum_vec.dot_acc(&vec1,&vec2) /* my_accum_vec += dot(vec1,vec2)`  .. which would obviously be easier for inference?  certainly seems merit in something like that, along with maybe even "accumulating matrix multiplies" aswell